src/seedu/addressbook/commands/Command.java

```diff
@@ -63,6 +64,11 @@ protected ReadOnlyPerson getTargetPerson() throws IndexOutOfBoundsException {
         return relevantPersons.get(getTargetIndex() - DISPLAYED_INDEX_OFFSET);
     }
 
+    /**
+     * @return boolean whether command mutates the data
+     */
+    public abstract boolean isMutating();
```

Comment:
Good job on using an abstract method here :)

However, the header comment can be better replaced by this:
```
/**
  * Returns true if the command potentially mutates data
  */
```
rather than using the `@return` directly to improve readability.